src/content/Angular-CSharp/ClientApp/src/app/fetch-data/fetch-data.component.ts

```diff
@@ -8,8 +7,15 @@ import { HttpClient } from '@angular/common/http';
 export class FetchDataComponent {
   public forecasts: WeatherForecast[] = [];
 
-  constructor(http: HttpClient, @Inject('BASE_URL') baseUrl: string) {
-    http.get<WeatherForecast[]>(baseUrl + 'weatherforecast').subscribe(result => {
+  constructor(@Inject('BASE_URL') baseUrl: string) {
+    fetch(baseUrl + 'weatherforecast').then(response => {
+////#if (IndividualLocalAuth)
+      if (response.redirected && response.url.startsWith(`${baseUrl}Identity/Account/Login`)) {
+        window.location.href = `${baseUrl}Identity/Account/Login?ReturnUrl=/fetch-data`;
```

Comment:
This is ugly, but it seems better than just logging a JSON parsing error to the console. If you used the opaque bearer token auth with `MapIdentityApi`, you'd get a 401 without a redirect, but Identity cookie auth redirects, and fetch follows redirects.

Reply:
Hoist the login path or maybe whole account path to a variable to avoid the path repetition?